pkg/cmd/pipeline/start.go

```diff
@@ -352,9 +354,51 @@ func (opt *startOptions) getInput(pipeline *v1alpha1.Pipeline) error {
 		}
 	}
 
+	if opt.UseParamDefaults && !opt.Last && opt.UsePipelineRun == "" {
+		if err = opt.includeDefaultParams(pipeline); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// Include the default in opts.Params
+func (opt *startOptions) includeDefaultParams(pipeline *v1alpha1.Pipeline) error {
+	//Creating a slice of string to contain the param.Name already included opt.Params through -p or --param option
+	var optParamNames = make([]string, 0)
+	for _, optParam := range opt.Params {
+		optParamNames = append(optParamNames, strings.Split(optParam, "=")[0])
```

Comment:
This would be better implemented as a map to not have to iterate over the whole slice each time with contains.